delivery_schenker/models/delivery_carrier.py

```
@@ -341,20 +341,40 @@ def _schenker_pickup_dates(self, picking):
         ).isoformat()
         return {"pickUpDateFrom": date_from, "pickUpDateTo": date_to}
 
-    def _schenker_shipping_information_package(self, picking, package):
-        weight = package.shipping_weight or package.weight
+    def _schenker_shipping_information_product_volume(self, product, qty):
+        return product.volume * qty
+
+    def _schenker_shipping_information_package_volume(self, picking, package):
         # Volume calculations can be unfolded with stock_quant_package_dimension
         if hasattr(package, "volume"):
-            volume = round(package.volume, 2)
-        else:
-            volume = sum([q.quantity * q.product_id.volume for q in package.quant_ids])
+            return package.volume
+        return sum(
+            [
+                self._schenker_shipping_information_product_volume(
+                    q.product_id, q.quantity
+                )
+                for q in package.quant_ids
+            ]
+        )
+
+    def _schenker_shipping_information_round_volume(self, volume):
+        return round(volume, 2)
+
+    def _schenker_shipping_information_default_volume(self):
+        return 0.01
```

Not linked to your refactoring but is it really better to declare a fake volume instead of having an error?

I would not do that.

---

I would also prefer to throw an error.

---

if it's an Odoo UserError window it should be OK, because the error traceback when Schenker errors is unreadable to a common user

---

I removed the method and added the UserError into the method _schenker_get_total_shipping_volume